kitsune/wiki/tests/test_views.py

```diff
@@ -684,3 +685,170 @@ def test_it_works(self):
+                             is_approved=True, save=True)
+        # Make the revision as current revision of the translated document
+        trans_doc.current_revision = trans_rev
+        trans_doc.save()
+        # return the english document and translated document
+        return en_doc, trans_doc
+
+    def get_data_from_translated_document(self, header, create_doc_locale, req_doc_locale):
+        """Create documents and get provided locale document data"""
+        # Create a client with provided ACCEPT_LANGUAGE header information
+        client = Client(HTTP_ACCEPT_LANGUAGE=header)
+        # Create English and translated Document based on the create_doc_locale
+        en_doc, trans_doc = self.create_documents(create_doc_locale)
+        # Get the data of the requested locale version of the document
+        url = reverse('wiki.document', args=[en_doc.slug], locale=req_doc_locale)
+        response = client.get(url, follow=True)
+        eq_(200, response.status_code)
+        doc = pq(response.content)
+        doc_content = doc('#doc-content').text()
+        return doc_content
+
+    def test_header_locales_in_translated_list(self):
+        """
+        Test that the article is showing in the header locale
+
+        Test while the document is not translated to requested locale but translated to
+        one of the locale mentioned in ACCEPT_LANGUAGE Header
+        """
+        # A Client with es, fr, ja language choices in ACCEPT_LANGUAGE Header
+        header = 'es,fr;q=0.7,ja;q=0.3,'
+        # Create a Translated document in fr locale version.
+        # And try to get data of es locale vesion of document with above mentioned client
+        doc_content = self.get_data_from_translated_document(header=header,
+                                                             create_doc_locale='fr',
+                                                             req_doc_locale='es')
+        # As the document is not translated into es locale version, but in the client header
+        # there is fr locale and the document is translated into fr locale, the user will see
+        # fr version of the document.
+        en_content = 'This article is in English'
+        trans_content = 'This article is translated into fr'
+        assert en_content not in doc_content
+        assert trans_content in doc_content
+
+    def test_header_locales_not_in_translated_list(self):
+        """
+        Test the article is showing in the fallback locale of header locele
+
+        * The article is not translated into requested locale
+        * The article is not translated to any of the locale of ACCEPT_LANGUAGE header
+        * But we have a fallback locale for one of the locales of ACCEPT_LANGUAGE header
+           ** This fallback locale is mentioned in kitsune/settings.py
+        *** Test that the fallback locale is being used
+        """
+
+        # A Client with de, an, ja language choices in ACCEPT_LANGUAGE Header
```

Comment:
# Define a client with de, an, ja as locale choices in the ACCEPT_LANGUAGE header